src/commands/deploy/genezio.ts

```diff
@@ -230,6 +234,55 @@ export async function genezioDeploy(options: GenezioDeployOptions) {
         });
     }
 
+    if (configuration.services?.crons) {
+        const res = await getFunctions(deployClassesResult?.projectEnvId || "").catch(() => {
+            throw new UserError("Something went wrong while fetching the cron functions.");
+        });
+        const crons: CronDetails[] = [];
+        for (const cron of configuration.services.crons) {
+            let cronFound = false;
+            const yamlFunctionName = await evaluateResource(
+                configuration,
+                cron.function,
+                options.stage || "prod",
+                undefined,
+                undefined,
+            );
+            for (const deployedFunction of res.functions) {
+                if (deployedFunction.name === `function-${yamlFunctionName}`) {
+                    crons.push({
+                        name: cron.name,
+                        url: deployedFunction.cloudUrl,
+                        endpoint: cron.endpoint || "",
+                        cronString: cron.schedule,
+                    });
+                    cronFound = true;
+                    break;
+                }
+            }
+            if (!cronFound) {
+                throw new UserError(
+                    `Function ${yamlFunctionName} not found in the deployed functions. Please make sure the function is deployed before adding it to the cron.`,
+                );
+            }
+        }
+        await syncCrons({
+            projectName: projectName,
+            stageName: options.stage || "prod",
+            crons: crons,
+        }).catch(() => {
+            throw new UserError("Something went wrong while syncing the cron jobs.");
+        });
+    } else {
+        await syncCrons({
+            projectName: projectName,
+            stageName: options.stage || "prod",
+            crons: [],
+        }).catch(() => {
+            throw new UserError("Something went wrong while syncing the cron jobs.");
+        });
```

Comment:
I am bit confused here - what should `syncCrons` do and why is it called twice?

Reply:
And also the errors messages should be improved - `Something went wrong` is not helping the user to find a solution - what he/she should do? Try to redeploy, try to debug, file an issue on github, contact support?

Reply:
Probably it's called twice to remove the existing crons, if there are any.